Fix progress for INSERT SELECT

azat opened the pull request:
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix progress for INSERT SELECT (before the fix for TCP protocol there was `written_rows`/`written_bytes` always 0 and for HTTP protocol all intermediate progress packets was with zero `written_rows`/`written_bytes`)

AVMusorin committed:
tests: cover progress for INSERT SELECT over HTTP

AVMusorin committed:
tests: cover progress for INSERT SELECT over TCP

azat committed:
Fix final progress for INSERT SELECT over TCP

Move sending of progress and profile events after calling
BlockIO::onFinish() since it will call on_finish callback, that will do
the final flush of progress (at least WriteProgress).

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
Fix written_bytes/written_rows in INSERT SELECT intermediate progress packets

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests: improve INSERT SELECT over HTTP coverage

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests: improve INSERT SELECT over TCP coverage

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

kssenii commented on tests/queries/0_stateless/02458_insert_select_progress_tcp.python:
```diff
@@ -0,0 +1,245 @@
+
+        byte = x & 0x7F
+        if x > 0x7F:
+            byte |= 0x80
```
🤔 
may be add a comment why this it is done this way?
not obvious calculations...

azat replied:
This was a copy-paste from `02010_lc_native.python`:

```
$ gg 'byte |= 0x80'
src/IO/VarInt.h:            byte |= 0x80;
src/IO/VarInt.h:            byte |= 0x80;
src/IO/VarInt.h:            byte |= 0x80;
tests/queries/0_stateless/02010_lc_native.python:            byte |= 0x80
```

azat replied:
And neither of this code has a comment for this, so I assume that it was considered as more or less generic?

kssenii replied:
Ok..

kssenii commented:
@azat looks like the new test is flaky, flaky check fails

azat committed:
tests: fix 02458_insert_select_progress_tcp flakiness

CI: https://s3.amazonaws.com/clickhouse-test-reports/42078/318042c1178f08f5c405a63a430089784fa19feb/stateless_tests_flaky_check__asan_.html

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
> looks like the new test is flaky, flaky check fails

Indeed, actually I've been waiting for this. Fixed.
